server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/LocalShardsBalancer.java

```diff
@@ -985,9 +1000,15 @@ private boolean tryRelocateShard(BalancedShardsAllocator.ModelNode minNode, Bala
                 if (allocationDecision.type() == Decision.Type.NO) {
                     continue;
                 }
+                // Skip moving the primary relocation when target has less primaries and is skewed due to
```

Comment:
Not quite following this, could you pls explain?

Reply:
Thanks for calling this out. This change basically is to prevent un-necessary primary relocations from maxNode to minNode or in other words avoid blind relocations when moving primary from maxNode -> minNode is not desirable. Updated the comment.